scribble-html-lib/scribble/html/resource.rkt

```diff
@@ -70,22 +69,23 @@
   (define file* (if (equal? file default-file) "" file))
   (define roots (current-url-roots))
   (define (find-root path mode)
-    (ormap (lambda (root+url+flags)
-             (let loop ([r (car root+url+flags)] [p path])
-               (if (pair? r)
-                 (and (pair? p) (equal? (car p) (car r))
-                      (loop (cdr r) (cdr p)))
-                 (case mode
-                   [(get-path) `(,(cadr root+url+flags) 
-                                 ,@p 
-                                 ,(if (and (equal? file* "")
-                                           (memq 'index (cddr root+url+flags)))
-                                      default-file
-                                      file*))]
-                   [(get-abs-or-true)
-                    (if (memq 'abs (cddr root+url+flags)) `("" ,@p) #t)]
-                   [else (error 'relativize "internal error: ~e" mode)]))))
-           roots))
+    (for/or ([root+url+flags (in-list roots)])
+      (let loop ([r (car root+url+flags)]
+                 [p path])
+        (if (pair? r)
+            (and (pair? p) (equal? (car p) (car r)) (loop (cdr r) (cdr p)))
+            (case mode
+              [(get-path)
+               `(,(cadr root+url+flags) ,@p
```

Comment:
@jackfirth I'd rather see 

```
(append (list (cadr root+url+flags)) 
        p 
        (list (if ...)))
```

here.

In general, I think quasiquote should only be preserved if there exists an item that is not unquote / unquote-splicing. If all of them are unquote / unquote-splicing, either it should be converted to a `list` or an `append`.

Reply:
I'd use a different rule for this. If I'm thinking of it as an sexpression, I'd use quasiquote and friends; if I'm thinking of it as a list, I'd use append and friends.

So maybe the quasiquote should be preserved by the tool and a user should make this judgment?

Reply:
I think it's pretty fuzzy in a lot of cases, so Resyntax is conservative and only takes a hard stance on the cases that seem clearly absurd to me like this:

```scheme
`(,a ,b ,c)
; just write `(list ...)` man, you're not fooling anyone
```

In the code you commented on, I would also prefer to avoid quotation for the same reason as you. But I dislike datum quotations generally and that's a much stronger stance than many Racketeers. So I don't think Resyntax should actively discourage it.